MaxText/decode_checkpoint.py

```diff
@@ -0,0 +1,123 @@
+"""
+ Copyright 2023 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ """
+
+# pylint: disable=g-bad-todo, abstract-method, consider-using-with, ungrouped-imports
+"""Generate a decode checkpoint with no opt_state from a training checkpoint."""
+
+import sys
+
+import checkpointing
+import jax
+import max_logging
+import max_utils
+import maxtext_utils
+import pyconfig
+
+from absl import app
+from etils import epath
+from jax.sharding import Mesh
+from jax import random
+from typing import Sequence
+from layers import Transformer
+
+def read_checkpoint(config, checkpoint_manager, is_decode=False):
```

Comment:
What does "is_decode" mean in this context?

Reply:
updated var naming to make it easier to comprehend.